Implemented Absence Requests

hpham17 opened the pull request:
I've completed the bulk of the project. I've tested it locally but I'm uncertain if current_user_id will save to an arbitrary Absence Request. (I wasn't able to GitHub authorize correctly with yours so I made my own key and used that to test locally). Please test and let me know how I did. Thank you.

chenhoward commented:
Hi Hubert.  Here is what needs to be updated:

- On the database add an approved column to an Absence.  Make it an integer column and on the model make that attribute an [enum] (http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html).  Let 0 = pending, 1 = Accepted, 2 = rejected.
- On model use [active record associations](http://guides.rubyonrails.org/association_basics.html) (User has_many :absences).
- Have two controllers:
  - Regular Controller: Allow index of all the users own absence requests (user = current_user), new, and create
  -  Admin Controller: A Absence controller scoped under the admin module.  Index sees all pending absences.  Has an edit and an update to approve/reject absences.  After approving or rejecting redirect back to admin absence index.

hpham17 commented:
HI Howard,

I've implemented the changes. Let me know if there's anymore.

warrenshen commented on app/controllers/admin/absences_controller.rb:
```diff
@@ -0,0 +1,32 @@
+  def reject
+    @absence = Absence.find(params[:id])
+    @absence.approved = 2
+      if @absence.save
```
Indentation a little off here!

warrenshen commented on app/controllers/admin/student_applications_controller.rb:
```diff
@@ -28,7 +28,7 @@ def pend
       flash[:accept] = "Pended #{student_application.first_name}"
     else
-      flash[:error] = "Error occured"
+      flash[:error] = "Error occurred"
```
Nice.

warrenshen commented on app/views/admin/absences/show.html.slim:
```diff
@@ -0,0 +1,7 @@
+.container
+  .row
+    .col-md-12
+        h1 = "Absence Request for #{@user.name}"
```
Indentation a little off here (I think it should be two spaces instead of four from the `.col-md-12` above).

warrenshen commented on config/routes.rb:
```diff
@@ -54,6 +55,12 @@
 
+    resources :absences do
+      member do
+        patch "accept", to: "absences#accept", as: 'accept'
```
Why not `"accept"` for consistency with the double quotes?

warrenshen commented:
@hpham17 some small comments and you got merge conflicts!

hpham17 commented:
Thanks for the comments Warren! I've fixed those. I'm not sure where I'm merge conflicting..it cant be because I fixed a typo right? I haven't pulled since I started working on this in Dec.. should I pull now?

warrenshen commented:
@hpham17 to fix the merge conflicts:
- Pull the remote master with `git pull origin master`
- You should get some merge conflicts
- Search through your files for `>>>>>>>` (or read the console logs that'll tell you where the conflicts are)
- Fix the conflicts by choosing which changes should be kept
- Commit and push your local back up!

hpham17 commented:
I'm sorry for the late reply Warren & Howard. Been incredibly busy / had no internet. I have fixed the merge conflicts.

chenhoward commented on db/schema.rb:
```diff
@@ -11,11 +11,23 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
+<<<<<<< HEAD
```
Run rake db:migrate to fix this

chenhoward commented on app/controllers/absences_controller.rb:
```diff
@@ -0,0 +1,29 @@
+class AbsencesController < ApplicationController
+  def index
+    @absences = Absence.where(user_id: current_user.id)
```
You can just do 
```ruby
current_user.absences
```

chenhoward commented:
Merge conlicts :(
